File: Array-Level-0/removeDuplicatesFromSortedArray/removeDuplicates.js
function removeDuplicates(arr) {
  let j = 0;
  for (let i = 1; i < arr.length;i++) {
    if(arr[i] != arr[j]) {
      j++;
      arr[j] = arr[i];
    }
  }
  return j + 1; 
}

let array = [0, 0, 1, 1, 1, 2, 2, 3, 3, 4]
let unique = removeDuplicates(array)
console.log(`The unique number of elements are : ${unique}`);

File: Array-Level-0/removeDuplicatesFromSortedArray/removeDuplicates.cpp
#include <iostream>
#include <vector>

using namespace std;

int removeDuplicates(vector<int> arr) {
  int j = 0;
  for (int i = 1; i < arr.size(); i++) {
    if (arr[i] != arr[j]) {
      j++;
      arr[j] = arr[i];
    }
  }
  return j + 1;
}

int main() {
  vector<int> arr = {0, 0, 1, 1, 1, 2, 2, 3, 3, 4};
  int unique = removeDuplicates(arr);
  cout << "The number of unique elements are " << unique;
  return 0;
}